### PR TITLE
Add support for welcome messages in secure transcript screen

### PR DESCRIPTION
**Pull request by gersonnoboa**
Some work is done on the secure transcript view model screen and some on the chat view model screen because of MSG-483. This bug means that receiving and displaying the welcome message is done when the secure transcript view model is active, but the handling of the response is done when the chat view model is active.

MOB-2116

### CODE REVIEW

File: GliaWidgets/SecureConversations/ChatTranscript/SecureConversations.ChatWithTranscriptModel.swift [hunk 1]
```diff
@@ -252,6 +252,14 @@ extension SecureConversations {
         var engagementAction: EngagementViewModel.ActionCallback?
         var engagementDelegate: EngagementViewModel.DelegateCallback?
 
+        // Used to check whether custom card contains interactable metadata.
+        var isInteractableCard: ((MessageRenderer.Message) -> Bool)?
+        // Used to check whether custom card should be hidden.
```
Comment (yurii-glia) [inline GliaWidgets/SecureConversations/ChatTranscript/SecureConversations.ChatWithTranscriptModel.swift:257]: does it make sense to add these comments to doc (just add one more slash)?

**Review by yurii-glia — APPROVED** [review]

**Review by rasmustautsglia — APPROVED** [review]

**Review by EgorovEI — APPROVED** [review]

File: GliaWidgets/SecureConversations/ChatTranscript/SecureConversations.ChatWithTranscriptModel.swift (hunk 1 above)
Comment (gersonnoboa) [reply GliaWidgets/SecureConversations/ChatTranscript/SecureConversations.ChatWithTranscriptModel.swift]: Fixed 7a289ee6a4529fbf7ec93aafa69f51da69af8020

**Comment by gersonnoboa** [comment]
@igorkravchenko added a test for checking system and welcome messages. Welcome messages are nothing more than automated operator messages, so I can just test that operator messages come correctly from the web socket.

**Comment by gersonnoboa** [comment]
!squash

**Comment by gersonnoboa** [comment]
!merge

**Pull request merged**